### ISSUE TITLE
[Proposal] Standardizing and clean up math.py functionalities and scope

### ISSUE THREAD

**Issue by ooctipus** [isaac-sim/IsaacLab#2917]
### Proposal

```
def make_pose(pos: torch.Tensor, rot: torch.Tensor) -> torch.Tensor:
    """Creates transformation matrices from positions and rotation matrices.

    Args:
        pos: Batch of position vectors with last dimension of 3.
        rot: Batch of rotation matrices with last 2 dimensions of (3, 3).

    Returns:
        Batch of pose matrices with last 2 dimensions of (4, 4).
    """
    assert isinstance(pos, torch.Tensor), "Input must be a torch tensor"
    assert isinstance(rot, torch.Tensor), "Input must be a torch tensor"
    assert pos.shape[:-1] == rot.shape[:-2]
    assert pos.shape[-1] == rot.shape[-2] == rot.shape[-1] == 3
    pose = torch.zeros(pos.shape[:-1] + (4, 4), dtype=pos.dtype, device=pos.device)
    pose[..., :3, :3] = rot
    pose[..., :3, 3] = pos
    pose[..., 3, 3] = 1.0
    return pose


def unmake_pose(pose: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
    """Splits transformation matrices into positions and rotation matrices.

    Args:
        pose: Batch of pose matrices with last 2 dimensions of (4, 4).

    Returns:
        Tuple containing:
            - Batch of position vectors with last dimension of 3.
            - Batch of rotation matrices with last 2 dimensions of (3, 3).
    """
    assert isinstance(pose, torch.Tensor), "Input must be a torch tensor"
    return pose[..., :3, 3], pose[..., :3, :3]
```

Before introduction of above two functions, treatment for position(in vector form) and rotation(in either angle-axis, euler, quat, matrix) were done separately. We had functions that clearly states which input form it expects from and which output form it produces.

After introduction of above two functions, we opened the gates for directly treating poses without going to lower level of treating individual levels of position and rotation.

However, this should be done carefully for our code base to not go into a mess and confusions, and we have not yet being able to do so.

I propose below 2 points

1.Before poses were almost always a tensor of shape 7. now it is unclear if the poses is in pos_quat form or matrix4x4 form. And all poses utility are operating in Matrix4x4 form but not pos_quat form. *We need either better notation for which function deal with which form, like what we have done before poses, or a better standardization in which form do we insist on using.*

2. Our math.py is reaching over 2000 lines of code, for better readbility, operations on poses may deserve a separete folder for its conceptually higher level operations### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

### Acceptance Criteria

Add the criteria for which this task is considered **done**. If not known at issue creation time, you can add this once the issue is assigned.

- [ ] We should carefully map our or math.py evolution before implementations

**Comment by RandomOakForest** [comment]
Thanks for posting this proposal. We will review it.
